### PR TITLE
Fix wrong calculation of weapon Angle in demo player

### PR DESCRIPTION
**Pull request by user578**
Fix wrong calculation of weapon Angle in demo player, more here: http://forum.ddnet.tw/viewtopic.php?f=45&t=2717
and
Add *.dll to .gitignore (for Windows)

### CODE REVIEW

**Comment by cinaera** [comment]
Where does this values (`800` and `1600`) come from? `256.0f*pi` ?

**Comment by user578** [comment]
Yes. Sorry, I had to use pi, my bad.
It should be `256.0f*pi` and `256.0f*pi*2` instead of `800` and `1600`.

**Comment by H-M-H** [comment]
Well I tested it and it seems to work, maybe you could shortly explain what was wrong.

**Comment by H-M-H** [comment]
Looks fine now.

File: src/game/client/components/players.cpp [hunk 1]
```diff
@@ -417,6 +417,18 @@ void CPlayers::RenderPlayer(
...
+		if (Player.m_Angle > (256.0f * pi) && Prev.m_Angle < 0)
+		{
+			Player.m_Angle -= 256.0f * 2 * pi;
+			Angle = mix((float)Prev.m_Angle, (float)Player.m_Angle, IntraTick) / 256.0f;
+		}
+		else if (Player.m_Angle < 0 && Prev.m_Angle > (256.0f * pi))
+		{
+			Player.m_Angle += 256.0f * 2 * pi;
+			Angle = mix((float)Prev.m_Angle, (float)Player.m_Angle, IntraTick) / 256.0f;
+		}
```
Comment (heinrich5991) [inline src/game/client/components/players.cpp]: Not sure why you're doing it down here instead of near line 409.

Comment (H-M-H) [inline src/game/client/components/players.cpp]: Well like this it is just executed less often (this correction is not required if you are the local character). In all other cases it is applied.

Comment (heinrich5991) [inline src/game/client/components/players.cpp]: By that logic, you should have moved the assignemnt part of line 409 down here as well.

**Pull request merged**